jib:dockerBuild fails: CacheDirectoryNotOwnedException

**Description of the issue**:

Simply updating jib from 0.9.0 to 0.9.1 for a project without a `clean` results in failure due to a `CacheDirectoryNotOwnedException`.
```
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.1:dockerBuild (default-cli) on project first: Build to Docker daemon failed, perhaps you should check that '/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/app/target/jib-cache' is not used by another application or set the `useOnlyProjectCache` configuration: CacheDirectoryNotOwnedException -> [Help 1]
```

It led me to check the ownership and permissions on `target/jib-cache`, which looked fine.

This error is thrown because the `target/jib-cache/.jib` file does not exist.

**Steps to reproduce**:

I think it would be as easy as:
```
$ mvn com.google.cloud.tools:jib-maven-plugin:0.9.0:dockerBuild
$ mvn com.google.cloud.tools:jib-maven-plugin:0.9.1:dockerBuild
```

**Environment**:
- MacOS 10.13.5
- Maven 3.5.3


**Log output**: <!-- If applicable, provide relevant log output -->

`mvn -e -X package`:
```
[INFO] --- jib-maven-plugin:0.9.1:dockerBuild (docker) @ first ---
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=187625, ConflictMarker.markTime=87648, ConflictMarker.nodeCount=107, ConflictIdSorter.graphTime=41493, ConflictIdSorter.topsortTime=57231, ConflictIdSorter.conflictIdCount=51, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=2145896, ConflictResolver.conflictItemCount=104, DefaultDependencyCollector.collectTime=300662461, DefaultDependencyCollector.transformTime=2551513}
[DEBUG] com.google.cloud.tools:jib-maven-plugin:jar:0.9.1:
[DEBUG]    com.google.http-client:google-http-client:jar:1.23.0:compile
[DEBUG]       com.google.code.findbugs:jsr305:jar:1.3.9:compile
[DEBUG]       org.apache.httpcomponents:httpclient:jar:4.0.1:compile
[DEBUG]          org.apache.httpcomponents:httpcore:jar:4.0.1:compile
[DEBUG]          commons-logging:commons-logging:jar:1.1.1:compile
[DEBUG]          commons-codec:commons-codec:jar:1.3:compile
[DEBUG]    org.apache.commons:commons-compress:jar:1.15:compile
[DEBUG]       org.objenesis:objenesis:jar:2.6:compile
[DEBUG]    com.google.guava:guava:jar:23.5-jre:compile
[DEBUG]       org.checkerframework:checker-qual:jar:2.0.0:compile
[DEBUG]       com.google.errorprone:error_prone_annotations:jar:2.0.18:compile
[DEBUG]       com.google.j2objc:j2objc-annotations:jar:1.1:compile
[DEBUG]       org.codehaus.mojo:animal-sniffer-annotations:jar:1.14:compile
[DEBUG]    com.fasterxml.jackson.core:jackson-databind:jar:2.9.2:compile
[DEBUG]       com.fasterxml.jackson.core:jackson-annotations:jar:2.9.0:compile
[DEBUG]       com.fasterxml.jackson.core:jackson-core:jar:2.9.2:compile
[DEBUG]    org.slf4j:slf4j-api:jar:1.7.25:compile
[DEBUG]    org.javassist:javassist:jar:3.22.0-GA:compile
[DEBUG]    org.apache.maven:maven-plugin-api:jar:3.5.2:compile
[DEBUG]       org.apache.maven:maven-model:jar:3.5.2:compile
[DEBUG]       org.apache.maven:maven-artifact:jar:3.5.2:compile
[DEBUG]       org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.3.3:compile
[DEBUG]          javax.enterprise:cdi-api:jar:1.0:compile
[DEBUG]             javax.annotation:jsr250-api:jar:1.0:compile
[DEBUG]       org.codehaus.plexus:plexus-utils:jar:3.1.0:compile
[DEBUG]       org.codehaus.plexus:plexus-classworlds:jar:2.5.2:compile
[DEBUG]    org.apache.maven:maven-core:jar:3.5.2:compile
[DEBUG]       org.apache.maven:maven-settings:jar:3.5.2:compile
[DEBUG]       org.apache.maven:maven-settings-builder:jar:3.5.2:compile
[DEBUG]          org.codehaus.plexus:plexus-interpolation:jar:1.24:compile
[DEBUG]          org.sonatype.plexus:plexus-sec-dispatcher:jar:1.4:compile
[DEBUG]             org.sonatype.plexus:plexus-cipher:jar:1.4:compile
[DEBUG]       org.apache.maven:maven-builder-support:jar:3.5.2:compile
[DEBUG]       org.apache.maven:maven-repository-metadata:jar:3.5.2:compile
[DEBUG]       org.apache.maven:maven-model-builder:jar:3.5.2:compile
[DEBUG]       org.apache.maven:maven-resolver-provider:jar:3.5.2:compile
[DEBUG]       org.apache.maven.resolver:maven-resolver-impl:jar:1.1.0:compile
[DEBUG]       org.apache.maven.resolver:maven-resolver-api:jar:1.1.0:compile
[DEBUG]       org.apache.maven.resolver:maven-resolver-spi:jar:1.1.0:compile
[DEBUG]       org.apache.maven.resolver:maven-resolver-util:jar:1.1.0:compile
[DEBUG]       org.apache.maven.shared:maven-shared-utils:jar:3.1.0:compile
[DEBUG]          commons-io:commons-io:jar:2.5:compile
[DEBUG]       org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.3:compile
[DEBUG]       com.google.inject:guice:jar:no_aop:4.0:compile
[DEBUG]          aopalliance:aopalliance:jar:1.0:compile
[DEBUG]       javax.inject:javax.inject:jar:1:compile
[DEBUG]       org.codehaus.plexus:plexus-component-annotations:jar:1.7.1:compile
[DEBUG]       org.apache.commons:commons-lang3:jar:3.5:compile
[DEBUG]    org.apache.maven.shared:maven-verifier:jar:1.6:compile
[DEBUG]       junit:junit:jar:3.8.2:compile
[DEBUG] Created new class realm plugin>com.google.cloud.tools:jib-maven-plugin:0.9.1
[DEBUG] Importing foreign packages into class realm plugin>com.google.cloud.tools:jib-maven-plugin:0.9.1
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>com.google.cloud.tools:jib-maven-plugin:0.9.1
[DEBUG]   Included: com.google.cloud.tools:jib-maven-plugin:jar:0.9.1
[DEBUG]   Included: com.google.http-client:google-http-client:jar:1.23.0
[DEBUG]   Included: com.google.code.findbugs:jsr305:jar:1.3.9
[DEBUG]   Included: org.apache.httpcomponents:httpclient:jar:4.0.1
[DEBUG]   Included: org.apache.httpcomponents:httpcore:jar:4.0.1
[DEBUG]   Included: commons-logging:commons-logging:jar:1.1.1
[DEBUG]   Included: commons-codec:commons-codec:jar:1.3
[DEBUG]   Included: org.apache.commons:commons-compress:jar:1.15
[DEBUG]   Included: org.objenesis:objenesis:jar:2.6
[DEBUG]   Included: com.google.guava:guava:jar:23.5-jre
[DEBUG]   Included: org.checkerframework:checker-qual:jar:2.0.0
[DEBUG]   Included: com.google.errorprone:error_prone_annotations:jar:2.0.18
[DEBUG]   Included: com.google.j2objc:j2objc-annotations:jar:1.1
[DEBUG]   Included: org.codehaus.mojo:animal-sniffer-annotations:jar:1.14
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-databind:jar:2.9.2
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-annotations:jar:2.9.0
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-core:jar:2.9.2
[DEBUG]   Included: org.javassist:javassist:jar:3.22.0-GA
[DEBUG]   Included: javax.enterprise:cdi-api:jar:1.0
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:3.1.0
[DEBUG]   Included: org.codehaus.plexus:plexus-interpolation:jar:1.24
[DEBUG]   Included: org.sonatype.plexus:plexus-sec-dispatcher:jar:1.4
[DEBUG]   Included: org.sonatype.plexus:plexus-cipher:jar:1.4
[DEBUG]   Included: org.apache.maven:maven-builder-support:jar:3.5.2
[DEBUG]   Included: org.apache.maven.resolver:maven-resolver-util:jar:1.1.0
[DEBUG]   Included: org.apache.maven.shared:maven-shared-utils:jar:3.1.0
[DEBUG]   Included: commons-io:commons-io:jar:2.5
[DEBUG]   Included: org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.3
[DEBUG]   Included: com.google.inject:guice:jar:no_aop:4.0
[DEBUG]   Included: aopalliance:aopalliance:jar:1.0
[DEBUG]   Included: org.codehaus.plexus:plexus-component-annotations:jar:1.7.1
[DEBUG]   Included: org.apache.commons:commons-lang3:jar:3.5
[DEBUG]   Included: org.apache.maven.shared:maven-verifier:jar:1.6
[DEBUG]   Included: junit:junit:jar:3.8.2
[DEBUG] Configuring mojo com.google.cloud.tools:jib-maven-plugin:0.9.1:dockerBuild from plugin realm ClassRealm[plugin>com.google.cloud.tools:jib-maven-plugin:0.9.1, parent: sun.misc.Launcher$AppClassLoader@55f96302]
[DEBUG] Configuring mojo 'com.google.cloud.tools:jib-maven-plugin:0.9.1:dockerBuild' with basic configurator -->
[DEBUG]   (f) allowInsecureRegistries = false
[DEBUG]   (f) project = MavenProject: ca.mt.jib.examples:first:0.0.1-SNAPSHOT @ /Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/app/pom.xml
[DEBUG]   (f) session = org.apache.maven.execution.MavenSession@76a82f33
[DEBUG]   (f) to = com.google.cloud.tools.jib.maven.JibPluginConfiguration$ToConfiguration@749ab7b4
[DEBUG]   (f) useOnlyProjectCache = false
[DEBUG] -- end configuration --
[DEBUG] Searching for main class... Add a 'mainClass' configuration to 'jib-maven-plugin' to improve build speed.
[DEBUG] Could not find a valid main class specified in 'maven-jar-plugin'; attempting to infer main class.
[WARNING] Base image 'gcr.io/distroless/java' does not use a specific image digest - build may not be reproducible
[INFO] 
[INFO] Containerizing application to Docker daemon as first:0.0.1-SNAPSHOT...
[DEBUG] Containerizing application with the following files:
[DEBUG] 	Classes:
[DEBUG] 		/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/app/target/classes/ca
[DEBUG] 	Resources:
[DEBUG] 		/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/app/target/classes/META-INF
[DEBUG] 	Dependencies:
[DEBUG] 		/Users/bsd/.m2/repository/com/google/guava/guava/21.0/guava-21.0.jar
[INFO] 
[DEBUG] TIMING	Building image to Docker daemon
[DEBUG] RUNNING	Building image to Docker daemon
[DEBUG] TIMED	Building image to Docker daemon : 1.04 ms
[DEBUG] Building image to Docker daemon : 1.04 ms
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.617 s
[INFO] Finished at: 2018-06-29T10:11:17-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.1:dockerBuild (docker) on project first: Build to Docker daemon failed, perhaps you should check that '/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/app/target/jib-cache' is not used by another application or set the `useOnlyProjectCache` configuration: CacheDirectoryNotOwnedException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.1:dockerBuild (docker) on project first: Build to Docker daemon failed, perhaps you should check that '/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/app/target/jib-cache' is not used by another application or set the `useOnlyProjectCache` configuration
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Build to Docker daemon failed, perhaps you should check that '/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/app/target/jib-cache' is not used by another application or set the `useOnlyProjectCache` configuration
    at com.google.cloud.tools.jib.maven.BuildDockerMojo.execute (BuildDockerMojo.java:116)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: com.google.cloud.tools.jib.cache.CacheDirectoryNotOwnedException
    at com.google.cloud.tools.jib.cache.Caches$Initializer.ensureOwnership (Caches.java:53)
    at com.google.cloud.tools.jib.cache.Caches$Initializer.init (Caches.java:103)
    at com.google.cloud.tools.jib.builder.BuildSteps.run (BuildSteps.java:187)
    at com.google.cloud.tools.jib.frontend.BuildStepsRunner.build (BuildStepsRunner.java:187)
    at com.google.cloud.tools.jib.maven.BuildDockerMojo.execute (BuildDockerMojo.java:112)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Additional Information**: <!-- Any additional information that may be helpful -->


<!-- Thanks for contributing! -->

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/frontend/BuildStepsRunner.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/frontend/BuildStepsRunner.java
@@ -238,8 +238,16 @@ public class BuildStepsRunner {
       throw new BuildStepsExecutionException(helpfulSuggestions.none(), ex);
 
     } catch (CacheDirectoryNotOwnedException ex) {
-      throw new BuildStepsExecutionException(
-          helpfulSuggestions.forCacheDirectoryNotOwned(ex.getCacheDirectory()), ex);
+      String helpfulSuggestion =
+          helpfulSuggestions.forCacheDirectoryNotOwned(ex.getCacheDirectory());
+      CacheConfiguration applicationLayersCacheConfiguration =
+          buildSteps.getBuildConfiguration().getApplicationLayersCacheConfiguration();
+      if (applicationLayersCacheConfiguration != null
+          && ex.getCacheDirectory()
+              .equals(applicationLayersCacheConfiguration.getCacheDirectory())) {
+        helpfulSuggestion = helpfulSuggestions.forCacheNeedsClean();
+      }
+      throw new BuildStepsExecutionException(helpfulSuggestion, ex);
     }
   }
 }

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/frontend/HelpfulSuggestions.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/frontend/HelpfulSuggestions.java
@@ -70,6 +70,10 @@ public class HelpfulSuggestions {
     return suggest("make sure that the registry you configured exists/is spelled properly");
   }
 
+  public String forCacheNeedsClean() {
+    return suggest("run `clean` to clear your build cache");
+  }
+
   public String forCacheDirectoryNotOwned(Path cacheDirectory) {
     return suggest(
         "check that '"

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/frontend/BuildStepsRunnerTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/frontend/BuildStepsRunnerTest.java
@@ -25,6 +25,7 @@ import com.google.cloud.tools.jib.builder.SourceFilesConfiguration;
 import com.google.cloud.tools.jib.cache.CacheDirectoryCreationException;
 import com.google.cloud.tools.jib.cache.CacheDirectoryNotOwnedException;
 import com.google.cloud.tools.jib.cache.CacheMetadataCorruptedException;
+import com.google.cloud.tools.jib.configuration.CacheConfiguration;
 import com.google.cloud.tools.jib.registry.InsecureRegistryException;
 import com.google.cloud.tools.jib.registry.RegistryUnauthorizedException;
 import com.google.common.collect.ImmutableList;
@@ -277,6 +278,31 @@ public class BuildStepsRunnerTest {
   }
 
   @Test
+  public void testBuildImage_cacheDirectoryNotOwnedException_needsClean()
+      throws InterruptedException, ExecutionException, CacheDirectoryNotOwnedException,
+          CacheMetadataCorruptedException, IOException, CacheDirectoryCreationException {
+    Path expectedCacheDirectory = Paths.get("some/path");
+
+    CacheDirectoryNotOwnedException mockCacheDirectoryNotOwnedException =
+        Mockito.mock(CacheDirectoryNotOwnedException.class);
+    Mockito.when(mockCacheDirectoryNotOwnedException.getCacheDirectory())
+        .thenReturn(expectedCacheDirectory);
+    Mockito.doThrow(mockCacheDirectoryNotOwnedException).when(mockBuildSteps).run();
+
+    Mockito.when(mockBuildConfiguration.getApplicationLayersCacheConfiguration())
+        .thenReturn(CacheConfiguration.forPath(expectedCacheDirectory));
+
+    try {
+      testBuildImageStepsRunner.build(TEST_HELPFUL_SUGGESTIONS);
+      Assert.fail("buildImage should have thrown an exception");
+
+    } catch (BuildStepsExecutionException ex) {
+      Assert.assertEquals(TEST_HELPFUL_SUGGESTIONS.forCacheNeedsClean(), ex.getMessage());
+      Assert.assertEquals(mockCacheDirectoryNotOwnedException, ex.getCause());
+    }
+  }
+
+  @Test
   public void testBuildImage_cacheDirectoryNotOwnedException()
       throws InterruptedException, ExecutionException, CacheDirectoryNotOwnedException,
           CacheMetadataCorruptedException, IOException, CacheDirectoryCreationException {
@@ -287,6 +313,9 @@ public class BuildStepsRunnerTest {
     Mockito.when(mockCacheDirectoryNotOwnedException.getCacheDirectory())
         .thenReturn(expectedCacheDirectory);
     Mockito.doThrow(mockCacheDirectoryNotOwnedException).when(mockBuildSteps).run();
+
+    Mockito.when(mockBuildConfiguration.getApplicationLayersCacheConfiguration())
+        .thenReturn(CacheConfiguration.forPath(Paths.get("another/path")));
 
     try {
       testBuildImageStepsRunner.build(TEST_HELPFUL_SUGGESTIONS);

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/frontend/HelpfulSuggestionsTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/frontend/HelpfulSuggestionsTest.java
@@ -44,6 +44,9 @@ public class HelpfulSuggestionsTest {
         "messagePrefix, perhaps you should make sure that the registry you configured exists/is spelled properly",
         TEST_HELPFUL_SUGGESTIONS.forUnknownHost());
     Assert.assertEquals(
+        "messagePrefix, perhaps you should run `clean` to clear your build cache",
+        TEST_HELPFUL_SUGGESTIONS.forCacheNeedsClean());
+    Assert.assertEquals(
         "messagePrefix, perhaps you should check that 'cacheDirectory' is not used by another application or set the `useOnlyProjectCache` configuration",
         TEST_HELPFUL_SUGGESTIONS.forCacheDirectoryNotOwned(Paths.get("cacheDirectory")));
     Assert.assertEquals(

